mfr/extensions/image/templates/viewer.mako

```diff
@@ -1,4 +1,176 @@
+                    ## 2.   Images are displayed in its actual size. No issue for all supported
+                    ##      browsers.
+                    baseImage.wrap("<div></div>").parent().css("display", "inline-block");
+                } else {
+                    ## Quirks: Chrome has a flickering bug when images are wrapped with `<div>` or
+                    ## `<span>`.  During zoom the scroll bar keeps appearing and disappearing which
+                    ## causes the image to keep resizing.  Wrapping with `<p>` instead to solve this
+                    ## annoying issue.
+                    baseImage.wrap("<p></p>")
+                }
+
+                wrapped = true;
+            };
+
+            var mouseZoom = function (event, delta) {
+
+                ## Disable page scroll when the mouse cursor are above the image
+                event.preventDefault();
+
+                if (delta > 0) {
+                    magScale += magStep;
+                    magScale = magScale > maxScale ? maxScale : magScale
+                } else {
+                    magScale -= magStep;
+                    magScale = magScale < minScale ? minScale : magScale
+                }
+
+                var image = $(".zoomImage");
+                magHeight = magHeight === null ? parseInt(image.css("height"), 10) : magHeight;
+                magWidth = magWidth === null ? parseInt(image.css("width"), 10) : magWidth;
+
+                image.css({
+                    width: magWidth * magScale,
+                    height: magHeight * magScale
+                });
+            };
+
+            ## Quirks: Need to use an in-memory copy of the image to prevent the zoom image from
+            ##         moving around.  Without this in-memoery copy, the zoom image moves around
+            ##         horizontally and "centers" wherever the mouse cursor is.
+            $("<img>").attr("src", $(baseImage[0]).attr("src")).load(function () {
+
+                widthLimit = this.width;
+                heightLimit = this.height;
+
+                ## Quirks: Disable zoom and display instructions for images of height less than 200.
+                ## 1.   The original issue was when the height is less than 150 (not 200), the zoom
+                ##      image moves around and thus does not cover the base image fully. However,
+                ##      I cannot trigger this any more.
+                ## 2.   When the height is less than around 200 (a little bit less than 200), which
+                ##      is close to the height of the popover instruction when shown, the bottom
+                ##      part of the instruction block is cut off.  Users need to scroll down to see
+                ##      the full message.  In addition, the scroll bar overlaps with the question
+                ##      circle.
+                ## 3.   Images of height less than 200 are not worth zooming anyway.
+                if (heightLimit < minHeightToZoom) {
```

Comment:
In addition to the description at the definition of `minHeightToZoom`, added a detailed comments on why it is needed and how the threshold is selected.